cources.server/softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRequestHandler.java

```diff
@@ -27,7 +25,8 @@ public String getName() {
 		ParameterizedType type = (ParameterizedType) currentClass.getGenericSuperclass();
 		Class parameter = (Class) type.getActualTypeArguments()[0];
 		T data = mapper.convert((Map<String,Object>)msg.getData(),(Class<T>)parameter);*/
-		T data = CommonReflectionUtils.getGenericParameterClass(this.getClass(), 0);
+		Class dataClass = CommonReflectionUtils.getGenericParameterClass(this.getClass(), 0);
+		T data = mapper.convert((Map<String,Object>)msg.getData(),(Class<T>)dataClass);
```

Comment:
@roma2341 Why you create new mapper in this method? We already have module that provides mapper.